### PR TITLE
[#173432219] Send a specific message in case the bonus activation is running (or not)

### PR DESCRIPTION
**Pull request by AleDore**
added activity for bonus_processing check before sending EligibilityCheckConflict messages

### CODE REVIEW

File: utils/messages.ts [hunk 1]
```diff
@@ -47,9 +47,18 @@ Se vuoi, puoi fare subito una simulazione o la richiesta online sul sito di INPS
   EligibilityCheckConflict: () =>
     ({
       subject: "Abbiamo completato le verifiche sul tuo ISEE",
-      markdown: `INPS ha completato le verifiche sull’ISEE e risulta che qualcuno del tuo nucleo familiare abbia già richiesto il Bonus Vacanze.  
+      markdown: `INPS ha completato le verifiche sull’ISEE e risulta che qualcuno del tuo nucleo familiare ha già richiesto il Bonus Vacanze.  
 
 Il Bonus è in fase di attivazione. Ti manderemo un messaggio quando sarà attivo.
+`
+    } as MessageContent),
```
Comment (balanza) [inline utils/messages.ts:54]: Consider remove this cast using something like

```typescript
    MessageContent.decode({
      subject: "Abbiamo completato le verifiche sul tuo ISEE",
      markdown: `INPS ha completato le verifiche sull’ISEE e risulta che qualcuno del tuo nucleo familiare ha già richiesto il Bonus Vacanze.  

Il Bonus è in fase di attivazione. Ti manderemo un messaggio quando sarà attivo.
`
    }).getOrElseL(e => {
      throw new Error(readableReport(e));
    })
```

File: EligibilityCheckOrchestrator/index.ts [hunk 2]
```diff
@@ -187,18 +188,44 @@ export const handler = function*(
...
+        SendMessageActivityInput.encode({
+          checkProfile: false,
+          content: getMessage(
+            maybeMessageType.value,
+            eligibilityCheckResponse.validBefore
+          ),
+          fiscalCode: eligibilityCheckResponse.fiscalCode
+        })
+      );
+    }
```
Comment (balanza) [inline EligibilityCheckOrchestrator/index.ts:228]: We need some test to cover all introduced branches

File: utils/messages.ts (hunk 1 above)
Comment (AleDore) [reply utils/messages.ts:54]: Maybe we'll do this refactor into an other PR ;)

File: CheckBonusStatusRunningActivity/handler.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        false as ActivityResult,
+        () => {
+          return true as ActivityResult;
+        }
+      )
+    ).getOrElseL(() => {
+      throw new Error(
+        "Cannot get BonusProcessing status on CheckBonusStatusRunningActivity"
+      );
+    });
```
[the remark this answers is not in the document]
Comment (AleDore) [reply CheckBonusStatusRunningActivity/handler.ts]: done

File: EligibilityCheckOrchestrator/index.ts (hunk 2 above)
Comment (AleDore) [reply EligibilityCheckOrchestrator/index.ts:228]: done

**Comment by gunzip** [comment]
which kind of check / query ?

**Comment by AleDore** [comment]
Can we query for each fiscalCode if exists a `BonusProcessing` right?

**Comment by gunzip** [comment]
Yes but the query will return a result only for the applicant fiscal code and there's no guarantee that it's the same bonus relative to this family (you can apply for more than one bonus). so you have to:

- query if there's a bonus processing for each fiscal code
- get the details of each bonus that this query returns
- compute familyUID and check if it's the same of the bonus relative to this eligibility check

which is overkill :-)

**Comment by gunzip** [comment]
that's why we're going to create an index with https://github.com/pagopa/io-functions-bonus/pull/120

File: EligibilityCheckOrchestrator/index.ts [hunk 4]
```diff
@@ -59,6 +64,7 @@ export const getMessageType = (
     : none;
 };
 
+// tslint:disable-next-line: no-big-function
```
Comment (AleDore) [inline EligibilityCheckOrchestrator/index.ts:67]: temporary disabled this rule cause yield operator limits fp-ts usage for code compression

**Comment by gunzip** [comment]
@BurnedMarshal can you take a look at this ?

**Pull request merged**